### DISCUSSION TITLE
Trying to use enable3d but get type errors

### DISCUSSION THREAD

**Discussion by Thijs-van-Drongelen** [enable3d/enable3d#226] [Help / Questions] [answered]
Hello,

Im wondering if I'm doing something wrong. Im getting errors like these:

![image](https://user-images.githubusercontent.com/19927333/144712431-148b96f1-385e-4d84-9374-ebfa1c41a68c.png)

When I check node_modules, there is no src folder.

![image](https://user-images.githubusercontent.com/19927333/144712453-a6363e74-891c-4f4d-9870-7cd80cf2c89a.png)

I got a couple of other errors like these. But maybe I'm doing something wrong.

**Comment by Thijs-van-Drongelen** [comment]
I got enable3d working in my existing project.

I can dust off this account and send you a pull request with the fixes required to get it running. (I work daily with gitlab, but cant be that different)

scene3d.d.ts and water.d.ts have a broken import and text.d.ts + draw.d.ts have an error in their typing.

**Reply by yandeu** [reply]
Really? All these things are broken?

The latest version of enable3d depends on the [following dependencies](https://github.com/enable3d/enable3d#dependencies). Make sure you don't mix different versions.

For example: You can't use `three r135` if enable3d uses `three r133.1`.

Why not?

Three.js does not follow [semver](https://semver.org/). Upgrading from one three.js version to another breaks almost everytime something.

**Reply by Thijs-van-Drongelen** [reply]
I cant get it to work without changing:

**scene3d.d.ts:**

`get animationMixers(): import("threeGraphics/jsm/plugins/mixers").GetMixers;`

to 
    
`get animationMixers(): import("@enable3d/three-graphics/jsm/plugins/mixers").GetMixers;`

**text.d.ts**

`clone(): TextTexture; `

to

`clone(): this;`

**water.d.ts**

`import { Color } from '@enable3d/common/src/types';`

to

`import { Color } from '@enable3d/common/dist/types';`

and

**draw.d.ts**

`clone(): DrawTexture;` -> clone(): this; `

and

` clone(): DrawSprite;` ->  clone(): this;


 "three": "0.133.1",

**Reply by yandeu** [reply]
I see now what you mean. Please set `"skipLibCheck": true,` in your `tsconfig.json`.
Or try the [template](https://github.com/enable3d/threejs-ammojs-webpack-example).

**Comment by yandeu** [comment, marked as answer]
I converted this topic to a new issue.  Please follow it https://github.com/enable3d/enable3d/issues/227.
